File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = []
    
    for poncketmon in nums:
        # 보유한 폰켓몬 번호인지 확인
        if poncketmon not in answer:
            answer.append(poncketmon)
        
        # 보유 가능한 폰켓몬 수의 도달했는지 확인
        if len(answer) == len(nums) // 2:
            break
    
    return len(answer)

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    
    sort(nums.begin(), nums.end());
    
    // 중복되지 않은 번호 저장
    vector<int> poncketmon;
    int now_num = 0;
    for (int num : nums) {
        if (now_num != num) {
            poncketmon.push_back(num);
            now_num = num;
        }
    }
    
    answer = min(poncketmon.size(), nums.size() / 2);
    return answer;
}